dbt/include/snowflake/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -0,0 +1,18 @@
+{% macro snowflake__get_columns_spec_ddl() %}
+  {# loop through user_provided_columns to create DDL with data types and constraints #}
+  {% if config.get('constraints_enabled', False) %}
+    {%- set ns = namespace(at_least_one_check=False) -%}
+    {%- set user_provided_columns = model['columns'] -%}
+    (
+    {% for i in user_provided_columns -%}
+      {%- set col = user_provided_columns[i] -%}
+      {% set constraints = col['constraints'] -%}
+      {%- set ns.at_least_one_check = ns.at_least_one_check or col['check'] %}
+      {{ col['name'] }} {{ col['data_type'] }} {% for x in constraints %} {{ x or "" }} {% endfor %} {{ "," if not loop.last }}
+    {%- endfor %}
+  )
+    {%- if ns.at_least_one_check -%}
+      {{exceptions.warn("We noticed you have `check` configs, these are NOT compatible with Snowflake and will be ignored")}}
```

Comment:
Maybe link to https://docs.snowflake.com/en/sql-reference/constraints-overview.html ?

Maybe the wording can be similar to what they have?

>Snowflake supports defining and maintaining constraints, but does not enforce them, except for NOT NULL constraints, which are always enforced.